api/handlers/reporthandler.py

```diff
@@ -0,0 +1,257 @@
+import json
+import bson
+import dateutil
+import copy
+
+from .. import base
+from .. import config
+from .. import util
+
+log = config.log
+
+EIGHTEEN_YEARS_IN_SEC = 568036800
```

Comment:
Generally, I prefer these kind of "we made a const of a duration because we have to, sorry" declarations to be a call to the language's time functions, or at least the actual arithmetic. 

`18 * 365.25 * 24 * 60 * 60` is comprehensible in a way that `568036800` isn't :smile: